tests/playbooks/kind_crud.yml

```diff
@@ -0,0 +1,735 @@
+         metadata:
+           name: test-fedora
+           namespace: default
+      register: vm_facts
+
+    - name: Fail if VM delete task was 'changed'
+      assert:
+        that:
+        - "vm_facts is not changed"
+        - "vm_facts is success"
+        fail_msg: "Delete VM task did not return changed"
+
+- name: CRUD test for VMI
+  hosts: localhost
+  connection: local
+  tasks:
+    - name: Create test-fedora-vmi VMI
+      kubevirt_raw:
+        state: present
+        name: test-fedora-vmi
+        namespace: default
+        inline:
+          apiVersion: kubevirt.io/v1alpha2
+          kind: VirtualMachineInstance
+          metadata:
+            name: test-fedora-vmi
+            namespace: default
+          spec:
+            domain:
+              resources:
+                requests:
+                  memory: 512M
+              devices:
+                disks:
+                - volumeName: bootvolume
+                  name: bootdisk
+                  disk:
+                    bus: virtio
+                - name: cloudinitdisk
+                  volumeName: cloudinitvolume
+                  cdrom:
+                    bus: virtio
+            volumes:
+            - name: bootvolume
+              registryDisk:
+                image: kubevirt/fedora-cloud-registry-disk-demo:latest
+            - name: cloudinitvolume
+              cloudInitNoCloud:
+                userData: |
+                  #cloud-config
+                  hostname: test-fedora-vmi
+                  users:
+                  - name: kubevirt
+                    gecos: KubeVirt Project
+                    sudo: ALL=(ALL) NOPASSWD:ALL
+                    passwd: $6$JXbc3063IJir.e5h$ypMlYScNMlUtvQ8Il1ldZi/mat7wXTiRioGx6TQmJjTVMandKqr.jJfe99.QckyfH/JJ.OdvLb5/OrCa8ftLr.
+                    shell: /bin/bash
+                    home: /home/kubevirt
+                    lock_passwd: false
+                    ssh_authorized_keys:
+                    - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC+KMp1F4SrKitvgILf8Q2GQbcOslIg4sfqUOf+m832zwoS2Z2ndmpr/3F+NtLq56cTaEzK9IPBIkNLJfsPqRsToLBYZD+l4S5FfuZ2xwCxGBH+TVXApO+SiD6c84rmjOx47665iQvMHKL+n/5gVvSdYDuegNKnj4rRr/eHnG2yC4TVZl3oHI7TPOUJT+kKjSWP1UesWTZmmck39IaFSmorg31X7g9hJHwq9JEDQilcbnIyqDZKiH6Ju4GjOU8mqhazBFB4qu/QDDb25pDpPd2pQGBilGvm7gwJCVnyDk9YZQU7gYE734KLDf5tCKMmEQSjwFx2Tj9mfZveCIJkaj3T kubevirt
+      register: vmi_facts
+
+    - name: Fail if VMI create task was not 'changed'
+      assert:
+        that:
+        - "vmi_facts is changed"
+        fail_msg: "Create VMI task did not return changed"
+  
+    - name: wait for changes to appear
+      wait_for: timeout=15
+    
+    - name: Expose ssh port on test-fedora-vmi
+      k8s_service:
+        state: present
+        name: test-fedora-vmi-ssh-access
+        namespace: default
+        type: NodePort
+        ports:
+          - port: 22
+            protocol: TCP
+        selector:
+          kubevirt/vmi: test-fedora-vmi
```

Comment:
I think it should be `kubevirt/vm` here. Can you check whether it works with `vmi` as well?

Reply:
@pkliczewski IIRC, we usually expose VMI using `virtctl`
it resembles with other example as well
https://github.com/kubevirt/ansible-kubevirt-modules/blob/21b6a857a395f841c359d80ea78ee61366109178/tests/playbooks/k8s_service.yml#L16

Reply:
Hmm, this is interesting. I used example command from kubevirt [doc](https://kubevirt.io/user-guide/#/workloads/virtual-machines/expose-service):

> virtctl expose virtualmachineinstance vmi-ephemeral --name vmiservice --port 27017 --target-port 22

and when described created service I see `Selector:          kubevirt.io/vm=example-box`